downloads/swan-lake-release-notes/swan-lake-2201.5.0 /RELEASE_NOTE.md

```diff
@@ -103,8 +103,72 @@ To view other bug fixes, see the [GitHub milestone for Swan Lake 2201.5.0](https
+    ```java
+    public static Object convert(Object value, Type targetType);
+    ```
+- `getFunctionName()` and `getPathParameters()` APIs are introduced in the runtime `io.ballerina.runtime.api.Environment` class. They provide the Ballerina function name and path parameters associated with Java interop methods, respectively.
+
 ### Improvements
 
+#### Support for command line arguments of built-in subtypes
+
+The command-line arguments of Ballerina built-in subtypes are now supported to be parsed as operands.
+
+For the following `main` function,
+```ballerina
+public function main(byte byteVal, string:Char charVal, int:Signed8 int8Val) {
+}
+```
+the values can be passed through the command line arguments as follows.
+```
+bal run -- 1 b 33
+```
+#### Support ambiguous union type configurable variables
+
+When a structural value is provided for a configurable variable of a union type that includes more than one type descriptor, the inherent type used will be the first (leftmost) type descriptor, from which the value can be constructed.
+
+For example, given the following Ballerina code
+```ballerina
+type Person record {|
+    string name;
+    string city;
+|};
+
+configurable map<string>|Person configVar = ?;
+```
+and the `Config.toml` file contains the following,
+```toml
+configVar = {name = "Jack", city = "Colombo"}
+```
+`configVar` is instantiated with the inherent type `map<string>`.
+
+#### Support binding of resource functions to a generic native method
+
+A new way has been introduced to support the binding of any resource function to a generic native method, regardless of the resource path parameters. The generic native method should be defined with a `BArray` parameter, which represents all path parameters. To avoid errors due to overloaded methods, it is recommended to define parameter type constraints as well.
+
+For example, the following Ballerina resource function
+```ballerina
+isolated resource function get abc/[int p1]/[string p2]/[string p3]/[int ...p4] (string s) = @java:Method {
+        'class: "javalibs.app.App",
+        name: "getResource",
+        paramTypes: ["io.ballerina.runtime.api.values.BObject", "io.ballerina.runtime.api.values.BArray", "io.ballerina.runtime.api.values.BString"]
+    } external;
+```
+can be bound to the following Java method.
+```java
+public static void getResource(BObject client, BArray path, BString str) {
+}
+```
+
+#### Improvements in Runtime Java APIs
+
+- The `getType` runtime API which returns an `ObjectType` in `io.ballerina.runtime.api.values.BObject` class is now deprecated. A new API `getOriginalType` which returns the Type is introduced to return both the `ObjectType` and the type-reference type. We need to modify the previous `getType` API to return the Type after fixing the caching issue [#39850](https://github.com/ballerina-platform/ballerina-lang/issues/39850).
```

Comment:
```suggestion
- The `getType` runtime API which returns an `ObjectType` in `io.ballerina.runtime.api.values.BObject` class is now deprecated. A new API `getOriginalType` which returns the `Type` is introduced to return both the `ObjectType` and the type-reference type. We need to modify the previous `getType` API to return the `Type` after fixing the caching issue [#39850](https://github.com/ballerina-platform/ballerina-lang/issues/39850).
```

Reply:
Do we need to include the last sentence? It is like a reminder/to-do for us.

Reply:
Agreed. 
@warunalakshitha WDYT?

Reply:
+1